doozer/doozerlib/cli/release_gen_assembly.py

```diff
@@ -465,16 +467,22 @@ def _get_rhcos_container(self):
                 if not self.rhcos_version:
                     self._exit_with_error(f"Did not find RHCOS {self.primary_rhcos_tag} image for active group architecture: {arch}")
                 # get rhcos pullspecs for this arch from rhcos version value if not full arch nightly provided
-                if rhcos_el_major > 8:
-                    rhcos_build_url = f"{RHCOS_RELEASES_STREAM_URL}/{major_minor}-{rhcos_el_major}.{rhcos_el_minor}/builds/{self.rhcos_version}/{arch}/meta.json"
-                else:
-                    rhcos_build_url = f"{RHCOS_RELEASES_STREAM_URL}/{major_minor}/builds/{self.rhcos_version}/{arch}/meta.json"
-                rhcos_meta_json = requests.get(rhcos_build_url).json()
                 for tag in rhcos.get_container_configs(self.runtime):
-                    if tag.build_metadata_key not in rhcos_meta_json:
-                        self._exit_with_error(f'Did not find RHCOS "{tag.name}" image for active group architecture: {arch}')
-                    rhcos_image = rhcos_meta_json[tag.build_metadata_key]
-                    self.rhcos_by_tag[tag.name][arch] = f"{rhcos_image['image']}@{rhcos_image['digest']}"
+                    if self.runtime.group_config.rhcos.layered_rhcos:
+                        if not self.rhcos_node_id:
+                            self._exit_with_error(f"Did not find RHCOS {self.primary_rhcos_tag} node image id for architecture: {arch}")
+                        stdout, _ = exectools.cmd_assert(f"oc image info {tag.rhcos_index_tag.replace('node-image',f'{self.rhcos_node_id}-node-image')} --filter-by-os {go_arch_for_brew_arch(arch)} -o json")
+                        self.rhcos_by_tag[tag.name][arch] = f"quay.io/openshift-release-dev/ocp-v4.0-art-dev@{json.loads(stdout)['digest']}"
```

Comment:
Consider loading the output into a JSON on a separate line to enhance readability.